src/editor/operations.ts

```diff
@@ -205,8 +206,13 @@ export function formatRangeAsCode(range: Range): void {
             parts.push(partCreator.newline());
         }
     } else {
-        toggleInlineFormat(range, "`");
-        return;
+        if (hasBacktick) {
+            parts.unshift(partCreator.plain("``"));
+            parts.push(partCreator.plain("``"));
```

Comment:
I think this needs to be a bit smarter, someone could have selected ```foo``bar``` and this would not work for that, also please write a test to exercise this behaviour to prevent it regressing for the future

Reply:
You can reference https://github.com/matrix-org/matrix-react-sdk/blob/afa60acbf123e6e739bcd6fe409485fd338c8295/src/editor/deserialize.ts#L203 for how this is done for message edits

Reply:
Also as per the current test failure, it seems to have broken the toggling behaviour of this feature, it needs to be smarter in that it looks at the number of backticks at the edge of the selection to know whether to add or remove backticks, and count the backticks within the selection (not at the end) to know how many to wrap in

Reply:
> I think this needs to be a bit smarter, someone could have selected `` foo``bar `` and this would not work for that (it'd write `` ``foo``bar`` ``, also please write a test to exercise this behaviour to prevent it regressing for the future

I am not really good with tests, hopefully i'll use your help to get it done.